File: apps/api/app/schemas/clothing.py
"""衣物相关 Pydantic 数据契约。"""
import uuid
from datetime import datetime
from typing import Optional, Any
from pydantic import BaseModel, ConfigDict, Field


class ClothingItemBase(BaseModel):
    """衣物基础字段。"""
    category: str = Field(..., description="类别: top, bottom, coat, shoes, accessory")
    sub_category: str = Field(..., description="子类别: hoodie, jeans, sneakers 等")
    primary_color: str = Field(..., description="主色调")
    secondary_color: Optional[str] = Field(None, description="辅助色调")
    style: str = Field(..., description="风格: casual, formal, sporty, minimal 等")
    thickness: str = Field(..., description="厚度: thin, medium, thick")
    season: list[str] = Field(..., description="适用季节列表: spring, summer, autumn, winter")
    temp_min: float = Field(..., description="适宜最低温度 ℃")
    temp_max: float = Field(..., description="适宜最高温度 ℃")


class ClothingItemCreate(ClothingItemBase):
    """创建衣物时的请求体（不含图片，图片通过单独上传接口）。"""
    image_url: Optional[str] = Field(None, description="图片路径，由上传接口返回")
    raw_vlm_attributes: Optional[dict[str, Any]] = Field(None, description="VLM 识别原始 JSON")


class ClothingItemUpdate(BaseModel):
    """更新衣物时的请求体，所有字段可选。"""
    category: Optional[str] = None
    sub_category: Optional[str] = None
    primary_color: Optional[str] = None
    secondary_color: Optional[str] = None
    style: Optional[str] = None
    thickness: Optional[str] = None
    season: Optional[list[str]] = None
    temp_min: Optional[float] = None
    temp_max: Optional[float] = None


class ClothingItemResponse(ClothingItemBase):
    """衣物详情响应。"""
    id: uuid.UUID
    user_id: uuid.UUID
    image_url: str
    raw_vlm_attributes: Optional[dict[str, Any]] = None
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)


class ClothingListResponse(BaseModel):
    """衣物列表响应（含分页信息）。"""
    items: list[ClothingItemResponse]
    total: int
    page: int
    page_size: int


class ImageUploadResponse(BaseModel):
    """图片上传成功的响应。"""
    image_url: str
    filename: str

File: apps/web/src/types/clothing.ts
/** 衣物单品 */
export interface ClothingItem {
  id: string;
  user_id: string;
  image_url: string;
  category: string;
  sub_category: string;
  primary_color: string;
  secondary_color: string | null;
  style: string;
  thickness: string;
  season: string[];
  temp_min: number;
  temp_max: number;
  raw_vlm_attributes: Record<string, any> | null;
  created_at: string;
}

/** 衣物列表响应 */
export interface ClothingListResponse {
  items: ClothingItem[];
  total: number;
  page: number;
  page_size: number;
}

/** 创建衣物请求 */
export interface ClothingItemCreate {
  category: string;
  sub_category: string;
  primary_color: string;
  secondary_color?: string;
  style: string;
  thickness: string;
  season: string[];
  temp_min: number;
  temp_max: number;
  image_url?: string;
  raw_vlm_attributes?: Record<string, any>;
}

/** 更新衣物请求 */
export interface ClothingItemUpdate {
  category?: string;
  sub_category?: string;
  primary_color?: string;
  secondary_color?: string;
  style?: string;
  thickness?: string;
  season?: string[];
  temp_min?: number;
  temp_max?: number;
}

/** 图片上传响应 */
export interface ImageUploadResponse {
  image_url: string;
  filename: string;
}

/** VLM 视觉大模型识别结果 */
export interface VLMAnalysisResponse {
  category: string;
  sub_category: string;
  primary_color: string;
  secondary_color: string | null;
  style: string;
  thickness: string;
  season: string[];
  temp_min: number;
  temp_max: number;
  raw_vlm_attributes?: Record<string, any>;
  confidence_score?: number;
}

/** 搭配中单品详情 */
export interface OutfitItemDetail {
  id: string;
  category: string;
  sub_category: string;
  primary_color: string;
  secondary_color: string | null;
  style: string;
  thickness: string;
  image_url: string;
  temp_min: number;
  temp_max: number;
}

/** 评分维度明细 */
export interface ScoreBreakdown {
  weather_score: number;
  style_score: number;
  scene_score: number;
  color_score: number;
  overall_score: number;
}

/** 一套穿搭推荐结果 */
export interface OutfitRecommendation {
  outfit_id: string;
  items: OutfitItemDetail[];
  scores: ScoreBreakdown;
  reason: string;
}

/** 推荐请求参数 */
export interface RecommendationRequest {
  city: string;
  temperature?: number;
  weather_condition?: string;
  scene: string;
  target_style: string;
  top_k?: number;
}

/** 推荐响应结果 */
export interface RecommendationResponse {
  record_id?: string | null;
  city: string;
  current_temp: number;
  weather_condition: string;
  scene: string;
  target_style: string;
  total_candidates: number;
  recommendations: OutfitRecommendation[];
}

/** 穿搭方案实体 */
export interface Outfit {
  id: string;
  user_id: string;
  name: string;
  occasion: string;
  season: string;
  item_ids: string[];
  overall_score: number;
  created_at: string;
}

/** 推荐历史记录 */
export interface RecommendationRecord {
  id: string;
  user_id: string;
  city: string;
  current_temp: number;
  weather_condition: string;
  scene: string;
  target_style: string;
  candidate_outfits: any[];
  ai_reason: string;
  feedback_rating: number | null;
  created_at: string;
}

/** AI 模特试穿生图请求 */
export interface TryOnRequest {
  outfit_id: string;
  items: Array<{
    sub_category: string;
    primary_color: string;
    image_url?: string;
  }>;
  gender?: string;
  scene?: string;
  target_style?: string;
}

/** AI 模特试穿生图结果 */
export interface TryOnResponse {
  outfit_id: string;
  image_url: string;
  prompt: string;
  source: string;
}

